src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs

```diff
@@ -251,15 +251,17 @@ internal static MsalTokenResponse CreateFromManagedIdentityResponse(ManagedIdent
         {
             ValidateManagedIdentityResult(managedIdentityResponse);
 
-            long expiresIn = DateTimeHelpers.GetDurationFromNowInSeconds(managedIdentityResponse.ExpiresOn);
+            long expiresOn = long.TryParse(managedIdentityResponse.ExpiresIn, out long expiresIn)
+                ? expiresIn
+                : DateTimeHelpers.GetDurationFromNowInSeconds(managedIdentityResponse.ExpiresOn);
 
             return new MsalTokenResponse
             {
                 AccessToken = managedIdentityResponse.AccessToken,
-                ExpiresIn = expiresIn,
+                ExpiresIn = expiresOn,

```

Comment:
Why this change?

Reply:
Looks like you are calculating expiresIn from the managed identity response which sends back expiresOn. Rename it back to expiresIn?